vmm/src/vm.rs

```diff
@@ -731,15 +731,16 @@ impl Vm {
         }
     }
 
-    pub fn add_disk(&mut self, mut _disk_cfg: DiskConfig) -> Result<()> {
+    pub fn add_disk(&mut self, mut _disk_cfg: DiskConfig) -> Result<u32> {
         if cfg!(feature = "pci_support") {
             #[cfg(feature = "pci_support")]
             {
-                self.device_manager
+                let new_disk_id = self
```

Comment:
Minor, but it's relatively uncommon to bind Result directly. It'd be clearer to keep the `?` and have `Ok(new_disk_id)` down below.

Reply:
Agreed. Nice catch